src/Symfony/Component/AssetMapper/AssetMapperCompiler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper;
+
+use Symfony\Component\AssetMapper\Compiler\AssetCompilerInterface;
+
+/**
+ * Runs a chain of compiles intended to adjust the source of assets.
+ *
+ * @experimental
+ */
+class AssetMapperCompiler
+{
+    /**
+     * @param iterable<AssetCompilerInterface> $assetCompilers
+     */
+    public function __construct(private iterable $assetCompilers)
+    {
+    }
+
+    public function compile(string $content, MappedAsset $mappedAsset, AssetMapper $assetMapper): string
+    {
+        foreach ($this->assetCompilers as $compiler) {
+            if (!$compiler->supports($mappedAsset)) {
+                continue;
+            }
+
+            $content = $compiler->compile($content, $mappedAsset, $assetMapper);
```

Comment:
so there can be many on one content string?

Reply:
There's no use-case specifically for that at the moment, but yea - and I think that's a safe thing to allow.